sky/server/requests/executor.py

```diff
@@ -385,12 +380,32 @@ def process_request(executor: concurrent.futures.ProcessPoolExecutor):
     # We use executor instead of individual multiprocessing.Process to avoid
     # the overhead of forking a new process for each request, which can be about
     # 1s delay.
-    with concurrent.futures.ProcessPoolExecutor(
+    try:
+        executor = concurrent.futures.ProcessPoolExecutor(
             max_workers=max_parallel_size,
             initializer=executor_initializer,
-            initargs=(proc_group,)) as executor:
+            initargs=(proc_group,))
         while True:
             process_request(executor)
+    except KeyboardInterrupt:
```

Comment:
Seems that ideally we shouldn't use KeyboardInterrupt for the worker SIGTERM. It makes sense for request execution since all the core code is designed to work with KeyboardInterrupt, but here we can fully control the exception type we throw and catch. We could use a custom exception.
Super low priority, probably better to just merge this. Could add a TODO comment.

Reply:
Sure! My original intent is to handle Ctrl-C (in `--foreground` mode) and SIGTERM from parent process in a uniform way, but agree it is better to use a distinct exception for SIGTERM